platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/VirtualMap.java

```diff
@@ -225,10 +226,13 @@ VirtualRootNode<K, V> getRoot() {
 
     /**
      * Does a full rehash of the persisted leaves of the map.
-     * Delegates the call to {@link VirtualRootNode#fullLeafRehash()} which does the actual work.
+     * Delegates the call to {@link VirtualRootNode#fullLeafRehashIfNecessary()} which does the actual work.
+     *
+     * This method is going to be removed after 0.39.0 release because it shouldn't be called explicitly.
```

Comment:
Was this method ever used by any promoted HS version? If not, it can be just dropped without deprecating it first

Reply:
Okay, I'll take my chances then. If someone grabs this platform build before me, they will have compilation errors. But I'll do my best to be first.